packages/web-react/src/components/Modal/useModalDialogStyleProps.ts

```diff
@@ -0,0 +1,45 @@
+import { ElementType, CSSProperties } from 'react';
+import { ModalDialogCSSHeight, ModalDialogCSSHeightBreakpoints, ModalDialogProps } from '../../types';
+
+interface CustomizedHeightCSSProperties extends CSSProperties {
+  [key: string]: string | undefined | number;
+}
+
+const setCustomHeight = (
+  baseVarName: string,
+  propValue: ModalDialogCSSHeight | ModalDialogCSSHeightBreakpoints,
+): CustomizedHeightCSSProperties => {
+  const customizedHeightStyle: CustomizedHeightCSSProperties = {};
+  if (typeof propValue === 'object' && propValue !== null) {
+    Object.keys(propValue).forEach((key) => {
+      const suffix = key === 'mobile' ? '' : `-${key}`;
+      const propName = `--${baseVarName}${suffix}`;
+
+      customizedHeightStyle[propName] = propValue[key as keyof ModalDialogCSSHeightBreakpoints]?.toString();
+    });
+  } else {
+    const propName = `--${baseVarName}`;
+    customizedHeightStyle[propName as keyof CustomizedHeightCSSProperties] = propValue?.toString();
+  }
+
+  return customizedHeightStyle;
+};
+
+export function useModalDialogStyleProps<E extends ElementType>(props: ModalDialogProps<E>) {
+  const { height, maxHeight, ...otherProps } = props;
+
+  let customizedHeightStyle: CustomizedHeightCSSProperties = {};
+
+  if (height) {
+    customizedHeightStyle = { ...customizedHeightStyle, ...setCustomHeight('modal-dialog-height', height) };
```

Comment:
```suggestion
    customizedHeightStyle = setCustomHeight('modal-dialog-height', height);
```

The `customizedHeightStyle` is an empty object from few lines above, am I right?

Reply:
Or maybe you can check the value by the `setCustomHeight` function, e. g. return early an empty object if there is no value.

So you can then get rid of those conditions. And just call the functions.

```js
const customizedHeightStyle = { ...setCustomHeight('modal-dialog-height', height), ...setCustomHeight('modal-dialog-max-height', maxHeight) };
```